### ISSUE TITLE
[CS8602] False positive on property of generic interface: 'Dereference of a possibly null reference'

### ISSUE THREAD

**Issue by wsy** [dotnet/roslyn-analyzers#7402]
### Analyzer

**Diagnostic ID**: [CS8602](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/nullable-warnings#possible-dereference-of-null): `Dereference of a possibly null reference`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 8 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 8.0.400](https://dotnet.microsoft.com/download/dotnet/8.0)

### Steps To Reproduce

Sample code
```csharp
#nullable enable
public interface IData1<TId>
{
    TId Id { get; }
}

public interface IData2<TId>
{
    TId? Id { get; }
}

public class SampleClass<TId>
{
    private IQueryable<IData1<TId>> dataStorage1;
    private IQueryable<IData2<TId>> dataStorage2;
    public IData1<TId>? GetById1(TId id) => dataStorage1.SingleOrDefault(c => c.Id.Equals(id));
    public IData2<TId>? GetById2(TId id) => dataStorage2.SingleOrDefault(c => c.Id.Equals(id));
}
```

### Expected behavior
Method `GetById2` shall generate CS8602 because property `Id` has type `TId?`.
Method `GetById1` shall not generate CS8602 because property `Id` has type `TId`.

### Actual behavior
Both `GetById1` and `GetById2` will generate CS8602.
![image](https://github.com/user-attachments/assets/98e6e16a-4446-4823-b300-edadff91b0a1)
